newrelic/hooks/framework_django.py

```diff
@@ -12,48 +12,71 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import functools
+import logging
 import sys
 import threading
-import logging
-import functools
-
-from newrelic.packages import six
 
 from newrelic.api.application import register_application
 from newrelic.api.background_task import BackgroundTaskWrapper
 from newrelic.api.error_trace import wrap_error_trace
-from newrelic.api.function_trace import (FunctionTrace, wrap_function_trace,
-        FunctionTraceWrapper)
+from newrelic.api.function_trace import (
+    FunctionTrace,
+    FunctionTraceWrapper,
+    wrap_function_trace,
+)
 from newrelic.api.html_insertion import insert_html_snippet
-from newrelic.api.transaction import current_transaction
 from newrelic.api.time_trace import notice_error
+from newrelic.api.transaction import current_transaction
 from newrelic.api.transaction_name import wrap_transaction_name
 from newrelic.api.wsgi_application import WSGIApplicationWrapper
-
-from newrelic.common.object_wrapper import (FunctionWrapper, wrap_in_function,
-        wrap_post_function, wrap_function_wrapper, function_wrapper)
+from newrelic.common.coroutine import is_asyncio_coroutine, is_coroutine_function
 from newrelic.common.object_names import callable_name
+from newrelic.common.object_wrapper import (
+    FunctionWrapper,
+    function_wrapper,
+    wrap_function_wrapper,
+    wrap_in_function,
+    wrap_post_function,
+)
 from newrelic.config import extra_settings
 from newrelic.core.config import global_settings
-from newrelic.common.coroutine import is_coroutine_function, is_asyncio_coroutine
+from newrelic.packages import six
 
 if six.PY3:
     from newrelic.hooks.framework_django_py3 import (
-        _nr_wrapper_BaseHandler_get_response_async_,
         _nr_wrap_converted_middleware_async_,
+        _nr_wrapper_BaseHandler_get_response_async_,
     )
 
+# These middleware are not useful to wrap as they don't do anything particularly
+# interesting that could cause performance issues.
+MIDDLEWARE_DENY_WRAP = [
```

Comment:
In situations like this in the agent, we would typically make this into a `frozenset()` for performance reasons.